### PR TITLE
Items pick and drop functionality

### LINKED ISSUE #194: Items and inventory
An inventory system to be used with tasks etc.

Should have:
- Items visible in player hand
- Ability to drop and pickup items, with shortcut and a GUI icon
- Dropped items visible on floor, can be picked up
- Toggle-able based on other events
- One item at a time
- When holding an item, toggles off the kill mechanic

See https://github.com/opensuspect/design-document/blob/main/general_gameplay.md.

### PR DESCRIPTION
**Pull request by zapzoop0099**
This PR gives the player ability to pick and drop items 

**Features:**

- Player can pick items and drop it and this sync over network

- If the item is picked by an infiltrator killing is disabled

- Items will hover in air when not picked

**This pr doesn't include:-**

- Items related to tasks

- Yellow shader over the items when get close to it

- Any type of asset(It have just a small box as placeholder for item prop )

The map already contained some instanced scene of items so when it get reviewed i will remove them at last
It also doesn't any include good comments :( but if it stays here for long i will push them
Please also refer to pr #246  and merge the best one.

### CODE REVIEW

File: src/assets/main/main.gd [hunk 1]
```diff
@@ -276,3 +282,62 @@ func _apply_customizations(player: KinematicBody2D, player_data: Dictionary) ->
 	face_wear.texture = load(appearance["Face Wear"]["texture_path"])
 	hat_hair.texture = load(appearance["Hat/Hair"]["texture_path"])
 	mouth.texture = load(appearance["Mouth"]["texture_path"])
+
+
+
+func _input(event):
+	if Input.is_action_just_pressed("ui_pick"):
```
Comment (jngo102) [inline src/assets/main/main.gd:289]: Putting an `Input.is_action_just_pressed` in `_input` doesn't make use of `event`.
Try using `event.is_action_pressed` instead.

File: src/assets/main/main.gd [hunk 2]
```diff
@@ -15,6 +15,10 @@ var newnumber
 var spawn_pos = Vector2(0,0)
 var player_data_dict: Dictionary
 
+var hand_pos:Dictionary
+var pick_range:Dictionary
+var keystroke:int
```
Comment (jngo102) [inline src/assets/main/main.gd:20]: Is `keystroke` ever used? What's its purpose?

File: src/assets/main/main.gd [hunk 3]
```diff
@@ -276,3 +282,62 @@ func _apply_customizations(player: KinematicBody2D, player_data: Dictionary) ->
...
+	
+	
+func get_hand_pos(id):
+	return hand_pos.get(id)
+	
+func get_pick_range(id):
+	return pick_range.get(id)
+
+
+func enable_killing(id):
```
Comment (jngo102) [inline src/assets/main/main.gd:328]: I'm not sure the expensive tree parsing in `enable_killing` and `disable_killing` are necessary here when you could just use
```
if players[id].has_node("Infiltrator"):
    players[id].get_node("Infiltrator").enable_killing([true/false])
```

**Review by jngo102 — CHANGES_REQUESTED** [review]

**Comment by zapzoop0099** [comment]
Didn't match with community standards

**Pull request closed without merging**